python/plugins/processing/algs/gdal/ClipRasterByExtent.py

```diff
@@ -57,6 +57,8 @@ def __init__(self):
     def initAlgorithm(self, config=None):
 
         self.TYPES = [self.tr('Use Input Layer Data Type'), 'Byte', 'Int16', 'UInt16', 'UInt32', 'Int32', 'Float32', 'Float64', 'CInt16', 'CInt32', 'CFloat32', 'CFloat64']
+        if GdalUtils.version() >= 3070000:
```

Comment:
It's more complex, but ideally we would always expose the new type and then check the version number at runtime (in processAlgorithm) with something like:

    if self.TYPES[type] == 'Int8' and GdalUtils.version() < 3070000:
      raise QgsProcessingException('Int8 data types require GDAL version 3.7 or later')

This means the processing algorithms expose a stable api, and users get a descriptive error if the type can't be used. Otherwise a user sharing a model using the int8 types with another user who DOESN'T have them available will get unpredictable results.

Reply:
Understood, Nyall. I'll rework this.

Reply:
@nyalldawson  it took some time but I think I did it the way you were suggesting.